iceoryx_posh/test/moduletests/test_mepoo_segment_management.cpp

```diff
@@ -54,7 +53,7 @@ class SegmentManager_test : public Test
     }
     void TearDown() override
     {
-        iox::rp::BaseRelativePointer::unregisterAll();
+        iox::rp::UntypedRelativePointer::unregisterAll();
```

Comment:
This is also misleading, as it affects all registered segments, regardless of type (not just untyped i.e. void). I think it should be a free function in some suitable namespace. The question then is what this namespace should be.

Reply:
@MatthiasKillat I see your point. However, I would argue that from the concept of `RelativePointer` it is clear that by calling `unregisterAll()` all the pointer in the repository are deleted the typed and the untyped. I will make this more clear in the doxygen documentation.

Reply:
The repository is an implementation artifact unknown at this point. Of course it works either way, it is just about names (and organization), but it is more like `iox::rp::registerPtr(...);`, `ìox::rp::unregisterAll(...);` etc.

We can do this later if at all, it is not very important.